### ISSUE TITLE
Suggest change to using EventEmitter

### ISSUE THREAD

**Issue by webprofusion-chrisc** [pjsb/ionic2-upload#1]
Callbacks seem to be a less popular option for change notification in angular2, for components the preferred approach is to have component Output. Suggest that instead of having a callback, have an event emitter so that files can be an observable output on the component:

``` typescript
  @Output("filesChanged")
  public filesChanged: EventEmitter<string>;
```

then have filesAdded be something like

``` typescript
 public filesAdded(event: Event): void {
    let files: FileList = this.nativeInputBtn.nativeElement.files;

    this.filesChanged.next(files);
  }
```

reference from template as 

```
(change)="filesAdded($event)"
```

When used in an another page etc this would look like:

``` html
   <upload-button [btnStyle]="create" (filesChanged)="uploadFile($event)">Upload</upload-button>
```

and the page can then handle the file selection changes that way.

**Comment by pjsb** [comment]
Good point will add this to my code. I will commit my current solution at the end of day, the first commit was just the first Idea ;)
